app/assets/javascripts/discourse/app/lib/user-search.js

```diff
@@ -50,22 +50,29 @@ function performSearch(
     return;
   }
 
+  let data = {
+    term: term,
+    topic_id: topicId,
+    category_id: categoryId,
+    include_groups: includeGroups,
+    include_mentionable_groups: includeMentionableGroups,
+    include_messageable_groups: includeMessageableGroups,
+    groups: groupMembersOf,
+    topic_allowed_users: allowedUsers,
+    include_staged_users: includeStagedUsers,
+    last_seen_users: lastSeenUsers,
+    limit: limit,
+  };
+
+  // });
+  Object.keys(customUserSearchOptions).forEach((key) => {
+    data[key.replace(/([a-zA-Z])(?=[A-Z])/g, "$1_").toLowerCase()] =
```

Comment:
Minor note but it'll make it easier to read if the regexp stuff is extracted into a function that has a name to explain what the regexp is doing.